openedx_rocketchat/views.py

```diff
@@ -0,0 +1,88 @@
+import hashlib
+import logging
+
+from django.http import Http404
+from django.contrib.auth.decorators import login_required
+
+from openedx_rocketchat.edxapp_wrapper.edxmako_module import render_to_response
+from openedx_rocketchat.edxapp_wrapper.get_configuration_helpers import get_configuration_helpers
+from openedx_rocketchat.edxapp_wrapper.get_courseware import get_course_with_access, has_access
+from openedx_rocketchat.edxapp_wrapper.get_opaque_keys import get_course_from_string
+from openedx_rocketchat.edxapp_wrapper.get_student_models import get_course_enrollment, get_course_enrollment_manager
+
+from .utils import (
+    create_course_group,
+    initialize_api_rocket_chat,
+    get_rocket_chat_settings,
+    create_token,
+)
+
+LOG = logging.getLogger(__name__)
+ROCKET_CHAT_DATA = "rocket_chat_data"
+
+
+@login_required
+def rocket_chat_discussion(request, course_id):
+    """
+    This view renders the rocketchat tab template.
+    """
+    configuration_helpers = get_configuration_helpers()
+    course_enrollment = get_course_enrollment()
+    course_enrollment_manager = get_course_enrollment_manager()
+
+    course_key = get_course_from_string(course_id)
+
+    user = request.user
+    course = get_course_with_access(user, 'load', course_key)
+
+    is_enabled = configuration_helpers.get_value(
+        'ENABLE_ROCKET_CHAT_SERVICE',
+        course.other_course_settings.get('enable_rocketchat_tab')
+    )
+
+    if not is_enabled:
+        raise Http404
+
+    staff_access = has_access(user, 'staff', course)
+    user_is_enrolled = course_enrollment.is_enrolled(user, course.id)
+
+    key = hashlib.sha1("{}_{}".format(ROCKET_CHAT_DATA, user.username)).hexdigest()
+
+    context = {
+        'request': request,
+        'course': course,
+        'course_title': course.display_name_with_default,
+        'staff_access': staff_access,
+        'user_is_enrolled': user_is_enrolled,
+        'beacon_rc': key,
+        "users_enrolled": course_enrollment_manager().users_enrolled_in(course_key) if staff_access else None,
+        'rocket_chat_error_message': 'Rocket chat service is currently not available',
+    }
+
+    rocket_chat_settings = get_rocket_chat_settings()
+
+    api_rocket_chat = initialize_api_rocket_chat(rocket_chat_settings)
+
+    if api_rocket_chat:
```

Comment:
If the api_rocket_chat is None, what happen next in the template view? @andrey-canon

Reply:
the message 'Rocket chat service is currently not available' will be shown